app/scripts/modules/core/src/pipeline/config/stages/googleCloudBuild/GoogleCloudBuildStageForm.tsx

```diff
@@ -60,17 +68,60 @@ export class GoogleCloudBuildStageForm extends React.Component<
     }));
   };
 
+  private getCloudBuildTriggers = (): Array<Option<string>> => {
+    return this.props.gcbTriggers.map(trigger => ({
+      label: trigger.name,
+      value: trigger.id,
+    }));
+  };
+
+  private getTriggerTypes = (): Array<Option<string>> => {
+    return [
+      { value: 'branchName', label: 'Branch name' },
+      { value: 'tagName', label: 'Tag Name' },
+      { value: 'commitSha', label: 'Commit SHA' },
+    ];
+  };
+
   private getSourceOptions = (): Array<Option<string>> => {
     return [
       { value: buildDefinitionSources.TEXT, label: 'Text' },
       { value: buildDefinitionSources.ARTIFACT, label: 'Artifact' },
+      { value: buildDefinitionSources.TRIGGER, label: 'Trigger' },
     ];
   };
 
   private onSourceChange = (e: any): void => {
     this.props.formik.setFieldValue('buildDefinitionSource', e.target.value);
   };
 
+  private onTriggerTypeValueChange = (e: any): void => {
```

Comment:
```suggestion
  private onTriggerTypeValueChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
```

Looks like a bunch of the old code used `any` as the event type so totally understandable that you would copy this...but let's be as strict as possible with typing for new code :smile: